rust/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -175,11 +177,11 @@ where
                 );
 
                 if let Some(tx) = self.broadcast_sender.as_ref() {
-                    logs.iter().for_each(|(_, meta)| {
-                        if let Err(err) = tx.send(meta.transaction_id) {
+                    for (_, meta) in &logs {
```

Comment:
this could be parallelized but given how fast the channels get populated given they can reach 30k items, I think the receivers are the bottleneck here